pkg/peertls/peertls.go

```diff
@@ -142,12 +150,23 @@ func WriteChain(w io.Writer, chain ...*x509.Certificate) error {
 		return errs.New("expected at least one certificate for writing")
 	}
 
+	var extErrs []error
 	for _, c := range chain {
 		if err := pem.Encode(w, NewCertBlock(c.Raw)); err != nil {
 			return errs.Wrap(err)
 		}
+
+		for _, e := range c.ExtraExtensions {
+			extBytes, err := asn1.Marshal(e)
+			if err != nil {
+				extErrs = append(extErrs, errs.Wrap(err))
+			}
+			if err := pem.Encode(w, NewExtensionBlock(extBytes)); err != nil {
+				extErrs = append(extErrs, errs.Wrap(err))
+			}
+		}
 	}
-	return nil
+	return utils.CombineErrors(extErrs...)
```

Comment:
And finally: `return extErrs.Finish()`, which will call internally `utils.CombineErrors`.